addons/website_crm_partner_assign_commission/models/account_invoice.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from odoo.tools import float_compare
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    reseller_id = fields.Many2one('res.partner', 'Reseller', help='This field is used to track the reseller in order to generate commisions')
+    commission_purchase_order_line_id = fields.Many2one('purchase.order.line', 'Reseller Purchase Order line')
+    commission_rate = fields.Float('Commission Rate', default=0,
+        help="Overrides all the commissions for this invoice, if set to 0 the commission_rate will be used")
+
+    @api.model
+    def _prepare_refund(self, invoice, date_invoice=None, date=None, description=None, journal_id=None):
+        res = super(AccountInvoice, self)._prepare_refund(invoice, date_invoice=date_invoice, date=date, description=description, journal_id=journal_id)
+        res.update({
+            'reseller_id': invoice.reseller_id.id,
+            'commission_purchase_order_line_id': invoice.commission_purchase_order_line_id.id,
+            'commission_rate': invoice.commission_rate,
+        })
+        return res
+
+    def _make_commissions(self):
+        """
+            Adds a line to a purchase order for reseller_id containing the due commissions for the invoice
+            The line is only added the first time we call this method (called on reconcile)
+        """
+        self.ensure_one()
+
+        if not self.reseller_id or self.commission_purchase_order_line_id:
```

Comment:
The `commission_purchase_order_line_id` is generated here, if it exists we don't want to generate another one.
We also don't want to try generating a line in case we don't have a reseller, but this part could probably be moved down in the `reconcile` method